crypto/asn1/x_pkey.c

```diff
@@ -75,6 +75,14 @@ X509_PKEY *X509_PKEY_new(void)
     if (ret->enc_algor == NULL || ret->enc_pkey == NULL)
         goto err;
 
+    ret->lock = CRYPTO_THREAD_lock_new();
```

Comment:
IMHO, you should move this block  a few lines upper, just after ret->reference = 1;
So, the if statement will be shorter.